src/python/pants/backend/jvm/tasks/bootstrap_jvm_tools.py

```diff
@@ -234,10 +237,22 @@ def _check_underspecified_tools(self, jvm_tool, targets):
         default option was provided to use for bootstrap.
         """.format(jvm_tool.key)))
 
+  # Force non-nailgun execution because nailgun will try to bootstrap itself via coursier,
```

Comment:
Since `IvyTaskMixin` didn't have this problem, how about fixing `CoursierMixin` to be a true mixin and moving its concrete `NailgunTask` (mixin!) base up to `CoursierResolve` which is a concrete leaf class. That should allow ditching this special-case forcing code which has this disadvantages of being confusing to the task configurer (I'm asking for nailgun but not getting it - why?) and keeping jarjat from running in a nailgun.

Reply:
Conformed to the structure for ivy resolve. `CoursierMixin` does not inherit `NailgunTask` anymore.